python/generate-short-url/main.py

```diff
@@ -0,0 +1,92 @@
+import json
+import requests
+
+
+def main(req, res):
+    # Required parameters
+    api_key = None
+    provider = None
+    url = None
+
+    try:
+        payload = json.loads(req.payload)
+        api_key = payload['api_key']
```

Comment:
Instead of taking this from payload, let's take it from `req.variables`. As mentioned in README, they should be called `BITLY_API_KEY` and `TINYURL_API_KEY`. Use depending on which provider request uses.